### PR TITLE
Column end condition

### PR DESCRIPTION
**Pull request by naterush**
# Description

Adds number of columns as a column ending condition.

# Testing

Use it. Write some more tests as well.

# Documentation

We need to add to Excel Range Import.

### CODE REVIEW

File: mitosheet/mitosheet/saved_analyses/step_upgraders/excel_range_import.py [hunk 1]
```diff
@@ -62,3 +62,66 @@ def upgrade_excel_range_import_1_to_2(step: Dict[str, Any], later_steps: List[Di
...
+            "range_imports": {
+                "type": 'range',
+                "df_name": str,
+                "value": str
+            } | {
+                "type": 'upper left corner value',
+                "df_name": str,
+                "value": str,
+                "end_condition": {'type': 'first empty cell'} | {'type': 'bottom left corner value', 'value': Any},
+                'column_end_condition': {'type': 'first empty cell'} | {'type': 'num columns', 'value': Any}
```
Comment (naterush) [inline mitosheet/mitosheet/saved_analyses/step_upgraders/excel_range_import.py:108]: value int

File: mitosheet/mitosheet/tests/step_performers/import_steps/test_excel_range_import.py [hunk 2]
```diff
@@ -126,30 +127,49 @@
...
+        ['A1:E3'],
+        [TEST_DF_5],
+        [{'type': 'upper left corner value', 'end_condition': {'type': 'first empty cell'}, 'column_end_condition': {'type': 'num columns', 'value': 3}, 'df_name': 'dataframe_1', 'value': TEST_DF_5.columns[0]}],
+        [TEST_DF_5.iloc[:, :3]],
+    ),
+    # Can't go where there are no data in the rows, so we put some fake data in J3
+    (
+        ['A1:E3', 'J3:K4'],
+        [TEST_DF_5, TEST_DF_1],
+        [{'type': 'upper left corner value', 'end_condition': {'type': 'first empty cell'}, 'column_end_condition': {'type': 'num columns', 'value': 10}, 'df_name': 'dataframe_1', 'value': TEST_DF_5.columns[0]}],
```
Comment (naterush) [inline mitosheet/mitosheet/tests/step_performers/import_steps/test_excel_range_import.py:170]: Add a test with not a first empty cell

**Comment by aarondr77** [comment]
When I update the select from Empty Cell to Num Columns, the selected value does not update. 
https://user-images.githubusercontent.com/18709905/228325910-19224cb9-75ce-40fa-80e0-d25110c90afd.mov

**Comment by aarondr77** [comment]
When I change the ending row condition, it also updates the Ending Column Condition 

https://user-images.githubusercontent.com/18709905/228326342-b595d5f8-c4f7-4bf7-9007-4c43596baada.mov

**Comment by aarondr77** [comment]
It is a bit confusing that the ending row conditions and the ending column conditions have different options. Don't need to address now, but I opened an issue here https://github.com/mito-ds/monorepo/issues/675

**Comment by aarondr77** [comment]
If ending row condition is set to bottom left corner then the ending column condition display does not update. Probably same issues as above ^

**Review by aarondr77 — COMMENTED** [review]
LGTM once bugs are fixed

File: mitosheet/mitosheet/step_performers/import_steps/excel_range_import.py [hunk 3]
```diff
@@ -55,13 +56,19 @@ def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Opti
...
                 end_condition = range_import['end_condition'] #type: ignore
-                if end_condition['type'] == EXCEL_RANGE_END_CONDITION_FIRST_EMPTY_VALUE:
-                    _range = get_table_range_from_upper_left_corner_value(file_path, sheet_name, range_import['value'])
-                elif end_condition['type'] == EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE:
-                    _range = get_table_range_from_upper_left_corner_value(file_path, sheet_name, range_import['value'], bottom_left_value=end_condition['value'])
-                else:
-                    raise Exception(f"End condition {end_condition} is not a valid type.")
+                column_end_condition = range_import['column_end_condition'] #type: ignore
+
+                assert end_condition['type'] in [EXCEL_RANGE_END_CONDITION_FIRST_EMPTY_VALUE, EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE]
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/mitosheet/step_performers/import_steps/excel_range_import.py:63]: We actually were doing it before - in practice, it's the same as raising an exception which we did before. Typing these things is almost impossibly with MyPy.
